src/movies.js

```diff
@@ -1,16 +1,145 @@
+/*
+const directors = getAllDirectors(movies)
+
+
+function cleanArray (dataDirectors){
+ 
+    const newDirectors = [...dataDirectors]
+    
+    //versión con filter 
+    const uniqueDirectors = newDirectors.filter((elem,index) => newDirectors.indexOf(elem) === index)
+    
+    //versión con new Set
+    const uniqueDirectors = [...new Set(newDirectors)]
+
+    return uniqueDirectors
+}
+
+cleanArray(directors)  
+
+*/
 
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
+function howManyMovies(dataMovies){
+    
+    const dramaBySpielberg = dataMovies.filter(elem => elem.director === 'Steven Spielberg' && elem.genre.indexOf('Drama') != -1)
 
+    return dramaBySpielberg.length
+
+}
+howManyMovies(movies)
 // Iteration 3: All rates average - Get the average of all rates with 2 decimals
+function ratesAverage(dataMovies){
+
+    const newDataMovies = [...dataMovies]
 
+    if (newDataMovies.length === 0) {
+        return 0
+    }
+
+    const movieWithRate = newDataMovies.filter(elem => elem.rate)
+
+    const averageRates = movieWithRate.reduce((acc, elem) => acc + elem.rate, 0)
+
+    const result = parseFloat((averageRates/movieWithRate.length).toFixed(2))
+
+    return result
+
+}
+ratesAverage(movies)
 // Iteration 4: Drama movies - Get the average of Drama Movies
+function dramaMoviesRate(dataMovies){
+
+    
+    const movieWithDrama = dataMovies.filter(elem => elem.genre.includes('Drama'))
+
+    if (movieWithDrama == 0){
+        return 0
+    }
+
+    const averageRates = movieWithDrama.reduce((acc, elem) => acc + elem.rate, 0)
+
+    const result = parseFloat((averageRates/movieWithDrama.length).toFixed(2))
+    
+    return result
+
+}
+dramaMoviesRate(movies)
+
 
 // Iteration 5: Ordering by year - Order by year, ascending (in growing order)
+function orderByYear(dataMovies){
+
+    const newDataMovies = [...dataMovies]
+    
+    const orderMovies = newDataMovies.sort((a,b) => (a.year - b.year))
+
+    return orderMovies
+
+}
+orderByYear(movies)
+
 
 // Iteration 6: Alphabetic Order - Order by title and print the first 20 titles
+function orderAlphabetically(dataMovies){
+
+    const newDataMovies = [...dataMovies]
+
+    const orderByAlpha = newDataMovies.sort((a,b) => (a.title.localeCompare(b.title)))
 
+    const result = orderByAlpha.slice(0, 20).map(elem => elem.title)
+
+    return result
+
+}
+orderAlphabetically(movies)
 // BONUS - Iteration 7: Time Format - Turn duration of the movies from hours to minutes
+function turnHoursToMinutes(dataMovies){
 
+    //const newDataMovies = [...dataMovies]
+    let newDataMovies = JSON.parse(JSON.stringify(dataMovies))
+    let result= []
+    let hours = []
+
+    const replaceHours = newDataMovies.map(elem => elem.duration.replace("h","").replace("min",""))     
+    
```

Comment:
Te falta contemplar la posibilidad de que haya películas en las cuales su propiedad _duration_ nos venga dada solo en minutos o solo en horas, como es un formato diferente, debes tratarlas de manera diferente a como lo haces dentro del for.